contracts/LiquidityWarsConfig.sol

```diff
@@ -236,10 +253,56 @@ contract LiquidityWarsConfig is Ownable {
         return usdRequiredAmount;
     }
 
-    function getAmountOfLpTokensRequired(
-        address _tokenAddress // TODO with this parameter
-    ) external view returns (uint256) {
-        return usdRequiredAmount; // Stub ! TODO: get the real value from router's query!
+    function getAmountOfLpTokensRequired(address _tokenAddress)
+        external
+        view
+        returns (uint256) 
+    {
+        // address factory = 0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f; // Uniswap Goerli
+        // address token0 = 0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984; //UNI token, Goerli
+        // address token1 = 0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6; //WETH token, Goerli
+
+        // address factory = 0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f; // Uniswap Mainnet
+        // address token0 = 0x514910771AF9Ca656af840dff83E8264EcF986CA; //LINK token, Mainnet
+        // address token1 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2; //WETH token, Mainnet
+
+        // address factory = 0xc35DADB65012eC5796536bD9864eD8773aBc74C4; // Sushiswap Mumbai
+        // address token0 = 0x326C977E6efc84E512bB9C30f76E30c160eD06FB; //LINK token, Mainnet
+        // address token1 = 0x9c3C9283D3e44854697Cd22D3Faa240Cfb032889; //WMATIC token, Mainnet
+
+        // address _pairContractAddress = UniswapV2Library.pairFor(factory, token0, token1);
+        // IUniswapV2Pair _pair = IUniswapV2Pair(_pairContractAddress);
+
+        IUniswapV2Pair _pair = IUniswapV2Pair(_tokenAddress);
+        uint totalSupply = IERC20(_tokenAddress).totalSupply();
+
+        // address token0 = _pair.token0();
+        // address token1 = _pair.token1();
+        // console.log("token0:", ERC20(token0).symbol());
+        // console.log("token1:", ERC20(token1).symbol());
+
+        (uint112 reserve0, uint112 reserve1,) = _pair.getReserves();
+        // console.log("reserve0:", reserve0, "LINK");
+        // console.log("reserve1:", reserve1, "MATIC");
+
+        int256 price0 = priceFeed0.getLatestPrice(); // LINK/MATIC
+        int256 price1 = priceFeed1.getLatestPrice(); // MATIC/USD
+        // console.log("price0:", uint256(price0), "LINK/MATIC (18 dec)");
+        // console.log("price1:", uint256(price1), "MATIC/USD (8 dec)");
+
+        uint256 reserve0USD = reserve0 * uint256(price0) * uint256(price1) / 1e18;
```

Comment:
Why divided by 1e18 if there is reserve0 * **18 dec * 8 dec** ? Shouldn't be more ?